transactor/src/main/scala/io/mediachain/datastore/PersistentDatastore.scala

```diff
@@ -0,0 +1,92 @@
+package io.mediachain.datastore
+
+import java.util.Random
+import java.util.concurrent.LinkedBlockingDeque
+import com.amazonaws.AmazonServiceException
+import org.slf4j.{Logger, LoggerFactory}
+import io.mediachain.multihash.MultiHash
+
+class PersistentDatastore(config: PersistentDatastore.Config)
+  extends BinaryDatastore with AutoCloseable with Runnable {
+  
+  val logger = LoggerFactory.getLogger(classOf[PersistentDatastore])
+  val dynamo = new DynamoDatastore(config.dynamo)
+  val rocks = new RocksDatastore(config.rocks)
+  val random = new Random
+  val maxBackoffRetry = 60 // second
+  val queue = new LinkedBlockingDeque[MultiHash]
+  val writer = new Thread(this, "PersistentDataStore#write")
```

Comment:
also maybe give the thread a unique name like `"PersistentDatastore#${this.hashCode}"`